apps/system/js/task_manager.js

```diff
@@ -440,25 +450,29 @@
         break;
 
       case 'select' :
-
-        if (this.position != card.position) {
-          // Make the target app, the selected app
-          this.position = card.position;
-          this.alignCurrentCard();
-        }
-
-        var self = this;
-        this.currentCard.element.addEventListener('transitionend',
-          function afterTransition(e) {
-            e.target.removeEventListener('transitionend', afterTransition);
-            self.exitToApp(card.app);
+        if (this.currentCard == card) {
+          this.exitToApp(card.app);
+        } else {
+          this.panToCard(card.position).then(() => {
```

Comment:
`panToCard` expects a card not a position :)
And it's a pretty important feature while we tweak the snapping physic.